File: chantier_manager_mobile/lib/models/vehicule.dart
class Vehicule {
  final int id;
  final String? immatriculation;
  final String? type;
  final String? libelle;
  final bool disponible;

  const Vehicule({
    required this.id,
    this.immatriculation,
    this.type,
    this.libelle,
    required this.disponible,
  });

  factory Vehicule.fromJson(Map<String, dynamic> json) {
    return Vehicule(
      id: json['id'] as int,
      immatriculation: json['immatriculation'] as String?,
      type: json['type'] as String?,
      libelle: json['libelle'] as String?,
      disponible: (json['disponible'] ?? true) as bool,
    );
  }

  Map<String, dynamic> toJson() => {
    "id": id,
    "immatriculation": immatriculation,
    "type": type,
    "libelle": libelle,
    "disponible": disponible,
  };
}

File: chantier-manager-api/src/main/java/istic/mmm/chantier_manager_api/entities/Vehicule.java
package istic.mmm.chantier_manager_api.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "vehicules")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String immatriculation;

    private String type;    // camion, utilitaire...

    private String libelle; // ex: "Camion benne 12T"

    private boolean disponible;

    //@ManyToMany(mappedBy = "vehicules")
    //@Builder.Default
    //private Set<Chantier> chantiers = new HashSet<>();
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chantier_id")
    @JsonIgnoreProperties({"vehicules"})
    private Chantier chantier;
}
